crates/rolldown_utils/src/ecma_script.rs

```diff
@@ -71,5 +66,5 @@ fn test_legitimize_identifier_name() {
   assert_eq!(legitimize_identifier_name("foo"), "foo");
   assert_eq!(legitimize_identifier_name("$foo$"), "$foo$");
   assert_eq!(legitimize_identifier_name("react-dom"), "react_dom");
-  assert_eq!(legitimize_identifier_name("111a"), "_11a");
+  assert_eq!(legitimize_identifier_name("111a"), "_111a");
```

Comment:
Before, if the first char is a invalid `start` we will replace it with `_`, but this broken the original naming, this pr fix that, just like rollup did.https://rollupjs.org/repl/?version=4.24.3&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMCU1QyUyMnNvbWUlMjBpbXBvcnQlNUMlMjIlMjBhcyUyMHNvbWVJbXBvcnQlMjAlN0QlMjBmcm9tJTIwJTVDJTIyX19mb28lNUMlMjIlNUNuZXhwb3J0JTIwJTdCJTIwc29tZUltcG9ydCUyMGFzJTIwJTVDJTIyc29tZSUyMGV4cG9ydCU1QyUyMiUyMCU3RCU1Q25leHBvcnQlMjAqJTIwYXMlMjAlNUMlMjJhbGwlMjB0aGUlMjBzdHVmZiU1QyUyMiUyMGZyb20lMjAlNUMlMjIxZm9vJTVDJTIyJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyaWlmZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlMjJqcXVlcnklMjIlM0ElMjIlMjQlMjIlMkMlMjIlMjQlMjIlM0ElMjJqcXVlcnklMjIlN0QlMkMlMjJpbnRlcm9wJTIyJTNBJTIyZGVmYXVsdCUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiU3RCU3RCU3RA==